### PR TITLE
#68-adding-amazon-to-extract

### PR DESCRIPTION
**Pull request by jk-george**
Closes: [Issue 68](https://github.com/users/gem09lo/projects/1/views/1?pane=issue&itemId=89987106&issue=gem09lo%7CPriceSlashTrack%7C68).

## Outcome:
- The extract.py script has a new function called scrape_pricing_process
- This function decides which scrape function to use in order to scrape the given URL (if it's Amazon/Steam it will use a respective function to scrape, if not it logs an error)
- Added a new function to scrape amazon website product pages and find the correct prices.
- Allowed schema.sql to have more characters in the product_name

### CODE REVIEW

File: pipeline/extract.py
Comment (gem09lo) [inline pipeline/extract.py:113]: Are you able to shorten this docstring? e.g. "Scrapes product, price and website information from Amazon."

File: pipeline/extract.py
Comment (gem09lo) [inline pipeline/extract.py:106]: can you add type hints

File: pipeline/extract.py
Comment (gem09lo) [inline pipeline/extract.py:149]: Can this also be shortened? e.g. "Scrapes product, price and website information from Steam."

File: pipeline/extract.py
Comment (gem09lo) [inline pipeline/extract.py:122]: Can this be renamed to something else - This variable name is specific to Steam products. Can rename it to product_name_element or something.

File: pipeline/extract.py
Comment (gem09lo) [inline pipeline/extract.py:124]: I didn't pick it up last time but following the code style guide we should avoid abbreviations.

File: pipeline/test_extract.py
Comment (gem09lo) [inline pipeline/test_extract.py:1]: Nice tests!

File: pipeline/test_extract.py
Comment (gem09lo) [inline pipeline/test_extract.py:6]: Good function names!

File: pipeline/test_extract.py
Comment (gem09lo) [inline pipeline/test_extract.py:7]: And descriptive docstrings

File: schema.sql
Comment (gem09lo) [inline schema.sql:24]: Nice spot

File: test_database.sql
Comment (gem09lo) [inline test_database.sql:37]: Good - more test data to work with. Im assuming this is products 12 and 13?

**Review by gem09lo — COMMENTED** [review]
Overall good code! Just a few minor changes for code quality. 
I'd want India to do a second review and approve it as I am not as familiar with beautiful soup as you guys are.

File: pipeline/extract.py
Comment (indiah444) [inline pipeline/extract.py:90]: Great function, makes sure we're scraping the right links

File: pipeline/extract.py
Comment (indiah444) [reply pipeline/extract.py:124]: Ah, I didn't realise that, I was the one who used abbreviations first, apologies!

**Review by indiah444 — CHANGES_REQUESTED** [review]
Great work so far, just a few small things should be changed for quality/consistency's sake (as mentioned in the comments before), but overall, the functionality of this is great.

File: pipeline/extract.py
Comment (jk-george) [reply pipeline/extract.py:113]: Done.

**Review by indiah444 — APPROVED** [review]
This all looks great, thank you for making these changes!

File: .github/workflows/pytest.yml
Comment (indiah444) [inline .github/workflows/pytest.yml:29]: Thank you very much for fixing this, this is fantastic!

Comment (jk-george) [reply .github/workflows/pytest.yml:29]: :)

File: dashboard.py
Comment (indiah444) [inline dashboard.py:128]: Apologies for using these abbreviations in the first place, thanks for fixing them! 😅

File: database/schema.sql
Comment (indiah444) [inline database/schema.sql:25]: Great fix to increase the length of the product name

**Comment by jk-george** [comment]
## Bug fixes:
The following bugs were fixed within this branch:
- [Issue 102](https://github.com/users/gem09lo/projects/1?pane=issue&itemId=90398300&issue=gem09lo%7CPriceSlashTrack%7C102)
- [Issue 103](https://github.com/users/gem09lo/projects/1/views/1?pane=issue&itemId=90398613&issue=gem09lo%7CPriceSlashTrack%7C103)
- [Issue 104](https://github.com/users/gem09lo/projects/1/views/1?pane=issue&itemId=90399070&issue=gem09lo%7CPriceSlashTrack%7C104)

## Technical Issue Found:
- [Technical Issue 105](https://github.com/users/gem09lo/projects/1/views/1?pane=issue&itemId=90399350&issue=gem09lo%7CPriceSlashTrack%7C105)

### Pull request will still move forward and be merged, even with this technical issue.

**Pull request merged**
